internal/webhooks/test/clusterclass_test.go

```diff
@@ -0,0 +1,421 @@
+// In this case deletion of the ClusterClass should succeed by the webhook as the update is compatible with the existing Clusters using the ClusterClass.
+func TestClusterWebhook_Succeed_Update(t *testing.T) {
+	defer utilfeature.SetFeatureGateDuringTest(t, feature.Gates, feature.ClusterTopology, true)()
+	g := NewWithT(t)
+
+	ns, err := env.CreateNamespace(ctx, "test-topology-cluster-reconcile")
+	g.Expect(err).ToNot(HaveOccurred())
+
+	// Create the objects needed for the integration test:
+	// - a ClusterClass with all the related templates
+	// - a Cluster using the above ClusterClass
+	clusterClass := builder.ClusterClass(ns.Name, "class1").
+		WithInfrastructureClusterTemplate(
+			builder.InfrastructureClusterTemplate(ns.Name, "inf").Build()).
+		WithControlPlaneTemplate(
+			builder.ControlPlaneTemplate(ns.Name, "cp1").
+				Build()).
+		WithWorkerMachineDeploymentClasses(
+			*builder.MachineDeploymentClass("NOT_USED").
+				WithInfrastructureTemplate(
+					builder.InfrastructureMachineTemplate(ns.Name, "OLD_INFRA").Build()).
+				WithBootstrapTemplate(
+					builder.BootstrapTemplate(ns.Name, "bootstrap1").Build()).
+				Build(),
+			*builder.MachineDeploymentClass("IN_USE").
+				WithInfrastructureTemplate(
+					builder.InfrastructureMachineTemplate(ns.Name, "infra1").Build()).
+				WithBootstrapTemplate(
+					builder.BootstrapTemplate(ns.Name, "bootstrap1").Build()).
+				Build()).
+		Build()
+
+	cluster := builder.Cluster(ns.Name, "cluster1").
+		WithLabels(map[string]string{clusterv1.ClusterTopologyOwnedLabel: ""}).
+		WithTopology(
+			builder.ClusterTopology().
+				WithVersion("v1.20.1").
+				WithClass("class1").
+				WithMachineDeployment(
+					builder.MachineDeploymentTopology("workers1").
+						WithClass("IN_USE").
+						Build(),
+				).
+				Build()).
+		Build()
+
+	// Create the ClusterClass in the API server.
+	g.Expect(env.Create(ctx, clusterClass)).To(Succeed())
+
+	// Create a Cluster using the ClusterClass.
+	g.Expect(env.Create(ctx, cluster)).To(Succeed())
+
+	// defer a function to clean up resources.
+	defer func() {
+		g.Expect(env.Delete(ctx, cluster))
+		g.Expect(env.Delete(ctx, clusterClass))
+		g.Expect(env.Delete(ctx, ns))
+	}()
+
+	// Retrieve the ClusterClass from the API server.
+	actualCluster := &clusterv1.ClusterClass{}
+	g.Expect(env.Get(ctx, client.ObjectKey{Namespace: ns.Name, Name: "class1"}, actualCluster)).To(Succeed())
+
+	// Remove MachineDeploymentClass "NOT_USED", which is not used by cluster1 from the clusterClass
+	actualCluster.Spec.Workers.MachineDeployments = []clusterv1.MachineDeploymentClass{
+		actualCluster.Spec.Workers.MachineDeployments[1],
+	}
+	// Change the template used in the ClusterClass to a compatible alternative (Only name is changed).
+	actualCluster.Spec.Infrastructure.Ref = objToRef(builder.InfrastructureClusterTemplate(ns.Name, "NEW_INFRA").Build())
```

Comment:
```suggestion
	actualCluster.Spec.Infrastructure.Ref.Name = "NEW_INFRA"
```
nit. Seems easier + we then probably don't need the comment